pkg/vetter/invalidserviceforjwtpolicy/vet.go

```diff
@@ -0,0 +1,160 @@
+	noServiceNoteSummary = "The authentication policy target service was not found in namespace '${namespace}'"
+	noServiceNoteMsg     = "The authentication policy '${policy}' in namespace '${namespace}' references the service" +
+		" '${service_target}', which does not exist in namespace '${namespace}'."
+)
+
+// Vetter implements Vetter interface
+type Vetter struct {
+	nsLister         listersvV1.NamespaceLister
+	svcLister        listersvV1.ServiceLister
+	authPolicyLister authv1alpha1.PolicyLister
+}
+
+func (v *Vetter) Vet() ([]*apiv1.Note, error) {
+	namespaces, err := v.nsLister.List(labels.Everything())
+	if err != nil {
+		glog.Error("Failed to retrieve namespaces: ", err)
+		return nil, err
+	}
+
+	var notes []*apiv1.Note
+	for _, ns := range namespaces {
+		nsAuthPolicies, err := v.authPolicyLister.Policies(ns.Name).List(labels.Everything())
+		if err != nil {
+			glog.Errorf("Failed to retrieve Authentication Policies for namespace: %s : %s", ns.Name, err)
+			return nil, err
+		}
+
+		nsServices, err := v.svcLister.Services(ns.Name).List(labels.Everything())
+		if err != nil {
+			glog.Errorf("Failed to retrieve Services for namespace: %s : %s", ns.Name, err)
+			return nil, err
+		}
+		nsServicesByName := createServiceLookup(nsServices)
+
+		for _, policy := range nsAuthPolicies {
+			policyNotes := createAuthPolicyNotes(policy, nsServicesByName)
+			notes = append(notes, policyNotes...)
+		}
+	}
+
+	return notes, nil
+}
+
+func createServiceLookup(services []*corev1.Service) map[string]*corev1.Service {
+	serviceLookup := make(map[string]*corev1.Service)
+	for _, s := range services {
+		serviceLookup[strings.ToLower(s.Name)] = s
+	}
+	return serviceLookup
+}
+
+func createAuthPolicyNotes(policy *v1alpha1.Policy, nsServiceLookup map[string]*corev1.Service) []*apiv1.Note {
+	var notes []*apiv1.Note
+	for _, o := range policy.Spec.GetOrigins() {
+		if o.GetJwt() != nil {
+			for _, t := range policy.Spec.GetTargets() {
+				targetSvc := nsServiceLookup[strings.ToLower(t.Name)]
+				if targetSvc == nil {
+					n := apiv1.Note{
+						Type:    noServiceNoteType,
+						Summary: noServiceNoteSummary,
+						Msg:     noServiceNoteMsg,
+						Level:   apiv1.NoteLevel_ERROR,
+						Attr: map[string]string{
+							"policy":         policy.Name,
+							"namespace":      policy.Namespace,
+							"service_target": t.Name,
+						},
+					}
+					notes = append(notes, &n)
+					continue
+				}
+
+				targetSvcIsValid := servicePortsContainAValidName(targetSvc)
+				if !targetSvcIsValid {
+					n := apiv1.Note{
+						Type:    invalidServiceNoteType,
+						Summary: invalidServiceNoteSummary,
+						Msg:     invalidServiceNoteMsg,
+						Level:   apiv1.NoteLevel_ERROR,
+						Attr: map[string]string{
+							"policy":         policy.Name,
+							"namespace":      policy.Namespace,
+							"service_target": targetSvc.Name,
+						},
+					}
+					notes = append(notes, &n)
+				}
+			}
+		}
+	}
+	return notes
```

Comment:
Need to compute IDs of the notes which is the SHA 256 of the strict, take a look at other vetters.